vertical-pod-autoscaler/e2e/updater.go

```diff
@@ -82,18 +92,76 @@ var _ = UpdaterE2eDescribe("Updater", func() {
 		permissiveMaxUnavailable := 7
 		// Creating new PDB and removing old one, since PDBs are immutable at the moment
 		setupPDB(f, "hamster-pdb-2", permissiveMaxUnavailable)
-		err = c.PolicyV1beta1().PodDisruptionBudgets(ns).Delete(pdb.Name, &metav1.DeleteOptions{})
+		err := c.PolicyV1beta1().PodDisruptionBudgets(ns).Delete(pdb.Name, &metav1.DeleteOptions{})
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
 		ginkgo.By(fmt.Sprintf("Waiting for pods to be restarted, sleep for %s", VpaEvictionTimeout.String()))
 		time.Sleep(VpaEvictionTimeout)
 		ginkgo.By("Checking enough pods were restarted.")
-		restartedCount, err := getPodSetChanges(c, d, podSet)
+		restartedCount := getPodSetChanges(c, getCurrentPodSetForDeployment(c, d), podSet)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		gomega.Expect(restartedCount >= permissiveMaxUnavailable).To(gomega.BeTrue())
 	})
 })
 
+func testRestartsPods(f *framework.Framework, controllerKind string) {
+	ginkgo.By(fmt.Sprintf("Setting up a hamster %v", controllerKind))
+	c := f.ClientSet
+	getCurrentPodList, podList := setupHamsterController(f, controllerKind, "100m", "100Mi", nil)
+
+	ginkgo.By("Setting up a VPA CRD")
+	setupVPA(f, "200m")
+
+	ginkgo.By("Waiting for pods to be restarted")
+	err := waitForPodSetChanged(c, getCurrentPodList, podList)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+type getCurrentPodListFunc func() (*apiv1.PodList, error)
+
+func setupHamsterController(f *framework.Framework, controllerKind, cpu, memory string, replicas *int32) (getCurrentPodListFunc, *apiv1.PodList) {
+	switch controllerKind {
+	case "Deployment":
+		d, podList := setupHamsterDeployment(f, cpu, memory, replicas)
+		getCurrentPodList := func() (*apiv1.PodList, error) {
+			return framework.GetPodsForDeployment(f.ClientSet, d)
```

Comment:
I wonder if we could use getHamsterPods() everywhere.
Even if it's not the most efficient nor specific function, it would make this case contain only one-liners and getCurrentPodList would be constant.

Reply:
Done.